chaco/toolbar_plot.py

```diff
@@ -51,12 +50,13 @@ def _bounds_changed(self, old, new):
         self.toolbar.do_layout(force=True)
         super(ToolbarPlot, self)._bounds_changed(old, new)
 
-    @on_trait_change('toolbar')
-    def _toolbar_changed(self, name, obj, old, new):
+    @observe('toolbar')
+    def _on_toolbar_changed(self, event):
```

Comment:
Maybe rename this to `_update_toolbar` if possible?

Also, it might be cleaner if we can create aliases `old, new = event.old, event.new` at the top of this method.